controllers/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -241,3 +255,38 @@ func NewTTR(tierTmpl *toolchainv1alpha1.TierTemplate, nsTmplTier *toolchainv1alp
 
 	return ttr
 }
+
+func (r *Reconciler) setStatusFailed(ctx context.Context, tmplTier *toolchainv1alpha1.NSTemplateTier, reason string, cause error) error {
+	if err := r.updateStatus(
+		ctx,
+		tmplTier,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.ConditionReady,
+			Status:  corev1.ConditionFalse,
+			Reason:  reason,
+			Message: cause.Error(),
+		}); err != nil {
+		logger := log.FromContext(ctx)
+		logger.Error(err, "unable to update NSTemplateTier condition")
+		return err
+	}
+	return cause
+}
+
+func (r *Reconciler) updateStatus(ctx context.Context, tmplTier *toolchainv1alpha1.NSTemplateTier, conditions ...toolchainv1alpha1.Condition) error {
+	var updated bool
+	tmplTier.Status.Conditions, updated = condition.AddOrUpdateStatusConditions(tmplTier.Status.Conditions, conditions...)
+	if !updated {
+		// Nothing changed
+		return nil
+	}
+	return r.Client.Status().Update(ctx, tmplTier)
+}
+
+func ReadyCondition() toolchainv1alpha1.Condition {
```

Comment:
should we add this in toolchain-common instead of it here as we have for different conditions?

Reply:
I was looking into it, but unless I've missed something , I see that we usually construct those in the controllers, for example see space_controller. But maybe I've missed where we keep them in toolchain-common. As far as I can see in common we have the test ones only.

Reply:
okay this can work.. if we don't use these condition outside of controller., it doesn't make sense to have it in toolchain-common.

i was actually referring to the conditions we set in toolchaincluster_controller.go https://github.com/codeready-toolchain/toolchain-common/blob/03e9ebab49c77d2c3906e452d9a16dd04d0f7444/controllers/toolchaincluster/toolchaincluster_controller.go#L130
and hence thought that similar conditions can be kept in toolchain-common controllers.

Reply:
for toolchaincluster we have the conditions along side with the controller, since that is a common controller. In this case the controller code is stored in host operator repo. 

But if we see any need for those conditions outside this repo we can always extract them later 👍